src/v2/api/services/ScheduleService.ts

```diff
@@ -94,4 +74,70 @@ export class ScheduleService {
         return indexOfLesson !== null;
       });
   }
+
+  async getEvent (id: string, week: number) {
+    const { startTime, endTime, ...event } = await this.eventRepository.find({ id });
+    const isEvent = !event.lessons.length;
+
+    week = week ?? await this.dateService.getCurrentWeek();
+
+    const { startDate } = await this.dateService.getCurrentSemester();
+    const startWeek = Math.ceil((startTime.getTime() - startDate.getTime()) / WEEK);
+    const endWeek = Math.ceil((endTime.getTime() - startDate.getTime()) / WEEK);
+
+    startTime.setDate(startTime.getDate() + (week - startWeek) * 7);
+    endTime.setDate(endTime.getDate() - (endWeek - week) * 7);
+
+    if (week > endWeek || week < startWeek) {
+      throw new InvalidWeekException();
+    }
+    if (event.period === Period.EVERY_FORTNIGHT && startWeek % 2 !== week % 2) {
+      throw new InvalidWeekException();
+    }
+
+    const eventInfoIndex = (week - startWeek) / (event.period === Period.EVERY_FORTNIGHT ? 2 : 1);
+    const eventInfo = event.eventInfo[eventInfoIndex].description;
+
+    let teachers = null, disciplineInfo = null, disciplineType: DisciplineType = null;
+    if (!isEvent) {
+      const discipline = await this.disciplineRepository.find({
+        disciplineTypes: {
+          some: {
+            lessons: {
+              some: {
+                eventId: event.id,
+              },
+            },
+          },
+        },
+      });
+      disciplineType = discipline.disciplineTypes.find((dt) => dt.id === event.lessons[0].disciplineTypeId);
+      teachers = discipline.disciplineTeachers
+        .filter((dt) => dt.roles.some((r) => r.disciplineTypeId === disciplineType.id))
+        .map(({ teacher }) => ({
```

Comment:
avoid mapping in services

Reply:
create better methods in mapper to use mapper